pkg/controller/csr/csr_controller.go

```diff
@@ -136,6 +150,12 @@ func allowedAssets(clusterDomain string) map[string]tlsAsset {
 			serviceaccountNamespace: rmonitor.TigeraPrometheusObjectName,
 			validDNSNames:           monitor.PrometheusTLSServerDNSNames(clusterDomain),
 		},
+		// The node-certs signing request originates from non-cluster hosts.
+		// To accommodate our customers' use of different non-cluster service accounts,
+		// we don't enforce checks on service account name and namespace.
+		render.NodeTLSSecretName: {
```

Comment:
Let's discuss the name once more. 
My current thinking is that it may be easy to have different names for nch-node-certs and regular ones, so we have the ability to have different validation. I imagine that different checks may apply (such as the service account, IP checks / DNS name checks, etc)